infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/util/EntryPoint.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.environment.util;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Objects;
+
+/** Represents an entry-point definition parsed from a string using the {@link EntryPointParser}. */
+public final class EntryPoint {
+
+  private final List<String> command;
+  private final List<String> arguments;
+
+  public EntryPoint(List<String> command, List<String> arguments) {
+    this.command = unmodifiableList(requireNonNull(command));
+    this.arguments = unmodifiableList(requireNonNull(arguments));
+  }
+
+  /** @return unmodifiable list representing the command of the entrypoint */
+  public List<String> getCommand() {
+    return command;
+  }
+
+  /** @return unmodifiable list representing the arguments of the entrypoint */
```

Comment:
"Unmodifiable" is deceptive here, since the list elements can be changed. Consider something like `ImmutableList.copyOf(arguments)` and similar for `getCommand()`

Reply:
Not sure if we're not overdoing it. The list cannot be changed through the returned reference, but of course it can change if something changes the list through the original reference passed to the ctor.

If that really is something we want to be covered for then I'd rather make the ctor package private and be only invoked from the parser, by which we can control the only code path that would hold such mutable reference.

Reply:
I mean your observation is of course 100% valid, but I'd be worried about it in some highly reusable piece of code where we could not assume the way it'd be used. IMHO, this is not such case.

Reply:
Ah sorry you are right, I had missed the `ImmutableList.copyOf(arguments);` above